pretixprint/app/src/main/java/eu/pretix/pretixprint/ui/BrotherRasterSettingsFragment.kt

```diff
@@ -0,0 +1,79 @@
+import org.jetbrains.anko.support.v4.defaultSharedPreferences
+
+class BrotherRasterSettingsFragment : SetupFragment() {
+
+    override fun onCreateView(
+            inflater: LayoutInflater,
+            container: ViewGroup?,
+            savedInstanceState: Bundle?
+    ): View {
+        val view = inflater.inflate(R.layout.fragment_brotherraster_settings, container, false)
+
+        val labelAdapter = ArrayAdapter(requireContext(), R.layout.list_item, BrotherRaster.Label.values())
+        (view.findViewById<TextInputLayout>(R.id.tilLabel).editText as? AutoCompleteTextView)?.setAdapter(labelAdapter)
+
+        val chosenLabelId = ((activity as PrinterSetupActivity).settingsStagingArea.get(
+            "hardware_${useCase}printer_label"
+        )) ?: defaultSharedPreferences.getString("hardware_${useCase}printer_label", "")
+        if (chosenLabelId?.isNotEmpty() == true) {
+            val chosenLabel = BrotherRaster.Label.values().find { it.name == chosenLabelId }.toString()
+            (view.findViewById<TextInputLayout>(R.id.tilLabel).editText as? AutoCompleteTextView)?.setText(chosenLabel, false)
+        }
+
+        val currentWaitAfterPage = ((activity as PrinterSetupActivity).settingsStagingArea.get(
+                "hardware_${useCase}printer_waitafterpage"
+        )) ?: defaultSharedPreferences.getString("hardware_${useCase}printer_waitafterpage", "2000")
+        view.findViewById<TextInputEditText>(R.id.teWaitAfterPage).setText(currentWaitAfterPage)
+
+        val currentRotate90 = ((activity as PrinterSetupActivity).settingsStagingArea.get(
+                "hardware_${useCase}printer_rotate90"
+        )?.toBoolean() ) ?: defaultSharedPreferences.getString("hardware_${useCase}printer_rotate90", "false")!!.toBoolean()
+        view.findViewById<SwitchMaterial>(R.id.swRotate90).isChecked = currentRotate90
+
+        view.findViewById<Button>(R.id.btnPrev).setOnClickListener {
+            back()
+        }
+        view.findViewById<Button>(R.id.btnNext).setOnClickListener {
+            val wap = view.findViewById<TextInputEditText>(R.id.teWaitAfterPage).text.toString()
+            val label = view.findViewById<TextInputLayout>(R.id.tilLabel).editText?.text.toString()
+            val rotate90 = view.findViewById<SwitchMaterial>(R.id.swRotate90).isChecked
+            if (TextUtils.isEmpty(wap)) {
+                view.findViewById<TextInputEditText>(R.id.teWaitAfterPage).error = getString(R.string.err_field_required)
+            } else if (TextUtils.isEmpty(label)) {
+                view.findViewById<TextInputEditText>(R.id.tilLabel).error = getString(R.string.err_field_required)
+            } else if (!TextUtils.isDigitsOnly(wap)) {
+                view.findViewById<TextInputEditText>(R.id.teWaitAfterPage).error = getString(R.string.err_field_invalid)
+            } else {
+                val mappedLabel = BrotherRaster.Label.values().find { it.toString() == label }!!.name
```

Comment:
I haven't tested the branch yet, but I think AutoCompleteTextVCiew does by default allow you to also enter invalid values, in which case this would cause a null pointer exception?

Reply:
Yeah, but thats why the AutoCompleteTextView has `android:inputType="none"` and `setText` is called with `filter: false` for selecting the last value. This seems to be the only current way to get a Material UI equivalent to a _Spinner_, but without a keyboard and the ability for the user to enter their own values.

Reply:
Ah, missed that, perfect